### DISCUSSION TITLE
Set up flux2 via terraform with a plain git repository

### DISCUSSION THREAD

**Discussion by rombert** [fluxcd/flux2#2326] [Q&A] [answered]
Hi,

I was looking to migrate my (simplistic) flux2 bootstrap to Terraform. Currently I have a Makefile that:

- creates the flux-system namespace
- creates a `sops-age` secret
- runs `flux bootstrap`, targetting a custom git repository over SSH

The public key setup is already handled.

I am looking at the terraform examples from https://registry.terraform.io/providers/fluxcd/flux/latest/docs, but they all seem to work with GitHub, and I'm not versed enough in Terraform to understand how to replicate this with a self-hosted git repository.

Are there any examples on bootstrapping flux with terraform + a generic git repository, or at least some hints in which direction to look?

**Comment by kingdonb** [comment, marked as answer]
The Terraform Provider does not run Flux Bootstrap, but it can emulate a similar process.

https://github.com/fluxcd/terraform-provider-flux/blob/62bd20c12920f63156153ed0842a2914fefc81f3/examples/github/main.tf#L139

If you take a look at the github example, there is this tls secret which is [loaded into a Kubernetes Secret](https://github.com/fluxcd/terraform-provider-flux/blob/62bd20c12920f63156153ed0842a2914fefc81f3/examples/github/main.tf#L109-L122) and it's connected with Flux.

This is the "hard part of bootstrap" as far as I can tell. This is the only part that involves interfacing with a GitHub API.

If you're not on GitHub, you obviously don't have that part.

We should have an example for non-github or in other words, generic git! I am not completely sure at a glance if any of these cover it now, but I don't think so: https://github.com/fluxcd/terraform-provider-flux/tree/main/examples – it wasn't obvious to me at a glance how to convert the GitHub example to generic Git, so I think this is needed.

You will have to arrange for the deploy key to be connected some other way, I'm pretty sure you're prepared for that already.

**Reply by rombert** [reply]
Thanks for the reply. I think I was looking at the terraform files from an incorrect definition. The current terraform provider should be usable with a plain Git repository. I'll come back with specific questions in case I have any.
